lib/tokenize.cpp

```diff
@@ -6267,30 +6271,46 @@ Token *Tokenizer::simplifyAddBracesPair(Token *tok, bool commandWithCondition)
             return tok;
         }
     }
+    // Skip labels
+    Token * tokStatement = tokAfterCondition;
+    while (true) {
+        if (Token::Match(tokStatement, "%name% :"))
+            tokStatement = tokStatement->tokAt(2);
+        else if (tokStatement->str() == "case") {
+            tokStatement = skipCaseLabel(tokStatement);
+            if (!tokStatement)
+                return tok;
+            if (tokStatement->str() != ":")
+                syntaxError(tokStatement);
+            tokStatement = tokStatement->next();
+        } else
+            break;
+        if (!tokStatement)
+            return tok;
+    }
     Token * tokBracesEnd=nullptr;
-    if (tokAfterCondition->str()=="{") {
+    if (tokStatement->str() == "{") {
         // already surrounded by braces
-        tokBracesEnd=tokAfterCondition->link();
-    } else if (Token::simpleMatch(tokAfterCondition, "try {") &&
-               Token::simpleMatch(tokAfterCondition->linkAt(1), "} catch (")) {
+        if (tokStatement != tokAfterCondition) {
+            // Move the opening brace before labels
+            Token::move(tokStatement, tokStatement, tokAfterCondition->previous());
```

Comment:
Is moving an existing token a good idea in general? Or it is better to add a new `{` token and remove the existing token? Or should I just add another pair of braces as was done before?

Reply:
That is not crystal clear. Neither approach is 100% safe. Feel free to move. At least you don't have to fear that you forget some flag or link etc then.

Reply:
OK, I will keep the move then.